tests/integration/targets/ndo_tenant_igmp_snooping_policy/tasks/main.yml

```diff
@@ -0,0 +1,401 @@
+          - nm_update_igmp_snooping_policy_again.current.enableFastLeaveControl == true
+          - nm_update_igmp_snooping_policy_again.current.enableQuerierControl == true
+          - nm_update_igmp_snooping_policy_again.current.lastMemberQueryInterval == 2
+          - nm_update_igmp_snooping_policy_again.current.igmpQuerierVersion == "v2"
+          - nm_update_igmp_snooping_policy_again.current.queryInterval == 100
+          - nm_update_igmp_snooping_policy_again.current.queryResponseInterval == 5
+          - nm_update_igmp_snooping_policy_again.current.startQueryCount == 1
+          - nm_update_igmp_snooping_policy_again.current.startQueryInterval == 25          
+          - nm_update_igmp_snooping_policy_again.current.uuid is defined
+
+    - name: Update a IGMP Snooping Policy name
+      cisco.mso.ndo_tenant_igmp_snooping_policy: &update_igmp_snooping_policy_name
+        <<: *update_igmp_snooping_policy
+        uuid: '{{ nm_update_igmp_snooping_policy.current.uuid }}'
+        name: ansible_test_igmp_snooping_policy_changed
+      register: nm_update_igmp_snooping_policy_name
+
+    - name: Assert IGMP Snooping Policy was updated
+      assert:
+        that:
+          - nm_update_igmp_snooping_policy_name is changed
+          - nm_update_igmp_snooping_policy_name.previous.name == "ansible_test_igmp_snooping_policy"
+          - nm_update_igmp_snooping_policy_name.current.name == "ansible_test_igmp_snooping_policy_changed"
+
+    # QUERY
+
+    - name: Create another IGMP Snooping Policy
+      cisco.mso.ndo_tenant_igmp_snooping_policy: &create_igmp_snooping_policy_2
+        <<: *create_igmp_snooping_policy
+        name: ansible_test_igmp_snooping_policy_2
+        description: "This is a test IGMP Snooping Policy 2"
+
+    - name: Query a IGMP Snooping Policy with template_name
+      cisco.mso.ndo_tenant_igmp_snooping_policy:
+        <<: *create_igmp_snooping_policy_2
+        state: query
+      register: query_one
+
+    - name: Query a IGMP Snooping Policy with template_name and UUID
+      cisco.mso.ndo_tenant_igmp_snooping_policy: &query_igmp_snooping_policy_uuid
+        <<: *create_igmp_snooping_policy_2
+        uuid: '{{ query_one.current.uuid }}'
+        name: '{{ fakevar | default(omit)}}'
+        state: query
+      register: query_one_uuid
+
+    - name: Query all IGMP Snooping Policy in the template
+      cisco.mso.ndo_tenant_igmp_snooping_policy:
+        <<: *mso_info
+        template: ansible_tenant_template
+        state: query
+      register: query_all
+
+    - name: Assert IGMP Snooping Policy was queried
+      assert:
+        that:
+          - query_one is not changed
+          - query_one.current.name == "ansible_test_igmp_snooping_policy_2"
+          - query_one_uuid is not changed
+          - query_one_uuid.current.name == "ansible_test_igmp_snooping_policy_2"
+          - query_all is not changed
+          - query_all.current | length == 2
```

Comment:
Should the test also check that the correct 2 policies are being returned?

Reply:
added tests